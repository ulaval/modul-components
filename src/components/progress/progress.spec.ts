import { createLocalVue, mount } from '@vue/test-utils';
import Vue, { VueConstructor } from 'vue';

import { renderComponent } from '../../../tests/helpers/render';
import ProgressPlugin, { MProgress } from './progress';

describe('MProgress', () => {
    let localVue: VueConstructor<Vue>;

    beforeEach(() => {
        localVue = createLocalVue();
        localVue.use(ProgressPlugin);
    });

    describe('bar mode', () => {
        it('should render correctly', () => {
            const pgr = mount(MProgress, {
                localVue: localVue
            });

            return expect(renderComponent(pgr.vm)).resolves.toMatchSnapshot();
        });
    });

    describe('indeterminate mode', () => {
        it('should render correctly', () => {
            const pgr = mount(MProgress, {
                localVue: localVue,
                propsData: {
                    indeterminate: true
                }
            });

            return expect(renderComponent(pgr.vm)).resolves.toMatchSnapshot();
        });

        it('should render correctly when circle', () => {
            const pgr = mount(MProgress, {
                localVue: localVue,
                propsData: {
                    circle: true
                }
            });

            return expect(renderComponent(pgr.vm)).resolves.toMatchSnapshot();
        });
    });
});
