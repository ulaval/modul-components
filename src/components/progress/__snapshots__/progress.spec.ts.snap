// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MProgress bar mode should render correctly 1`] = `
<div name="m--is" appear="" class="m-progress m--is-determinate  m--is-in-progress" style="height:6px;border-radius:3px;">
    <div class="m-progress__bar" style="width:0%;border-bottom-left-radius:3px;border-top-left-radius:3px;"></div>
</div>
`;

exports[`MProgress indeterminate mode should render correctly 1`] = `
<div name="m--is" appear="" class="m-progress m--is-indeterminate m--is-in-progress" style="height:6px;border-radius:3px;">
    <div class="m-progress__bar" style="width:0%;border-bottom-left-radius:3px;border-top-left-radius:3px;"></div>
</div>
`;

exports[`MProgress indeterminate mode should render correctly when circle 1`] = `
<div name="m--is" appear="" class="m-progress m--is-circle m--is-determinate  m--is-in-progress" style="height:100%;border-radius:initial;">
    <div class="m-progress__wrap" style="width:50px;height:50px;">
        <svg preserveAspectRatio="xMidYMid meet" focusable="false" viewBox="0 0 50 50" class="m-progress__draw" style="width:50px;height:50px;">
            <circle cx="50%" cy="50%" r="23" class="m-progress__background" style="stroke-dasharray:144.51326206513048px;stroke-width:4px;animation-name:m-progress-spinner-stroke-rotate-50;"></circle>
            <circle cx="50%" cy="50%" r="23" class="m-progress__circle" style="stroke-dashoffset:NaNpx;stroke-dasharray:144.51326206513048px;stroke-width:4px;animation-name:m-progress-spinner-stroke-rotate-50;"></circle>
        </svg>
    </div>
    <div class="m-progress__slot"></div>
</div>
`;
